src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/services/NodeService.kt

```diff
@@ -74,34 +76,53 @@ class NodeService(
         .switchIfEmpty(
           mono { authEvent }
             .map { ev ->
-              val timestamp =
-                OffsetDateTime
-                  .now() // FIXME this timestamp should the one coming from authorizationResultDto
-              ClosePaymentRequestV2Dto().apply {
-                paymentTokens = activatedEvent.data.paymentNotices.map { it.paymentToken }
-                outcome = transactionOutcome
-                idPSP = ev.data.pspId
-                paymentMethod = ev.data.paymentTypeCode
-                idBrokerPSP = ev.data.brokerName
-                idChannel = ev.data.pspChannelCode
-                this.transactionId = transactionId.toString()
-                totalAmount = (ev.data.amount.plus(ev.data.fee)).toBigDecimal()
-                fee = ev.data.fee.toBigDecimal()
-                timestampOperation = timestamp
-                additionalPaymentInformations =
-                  mapOf(
-                    "outcome_payment_gateway" to transactionOutcome.value,
-                    "authorization_code" to authorizationCode.orElseGet { "" },
-                    "tipoVersamento" to "CP",
-                    "rrn" to "123456789",
-                    "fee" to ev.data.fee.toBigDecimal().toString(),
-                    // bug CHK-1410: date formatted to yyyy-MM-ddTHH:mm:ss truncating millis
-                    "timestampOperation" to
-                      timestamp
-                        .toLocalDateTime()
-                        .truncatedTo(ChronoUnit.SECONDS)
-                        .format(DateTimeFormatter.ISO_LOCAL_DATE_TIME),
-                    "totalAmount" to (ev.data.amount.plus(ev.data.fee)).toBigDecimal().toString())
+              when (transactionOutcome) {
+                ClosePaymentRequestV2Dto.OutcomeEnum.KO ->
+                  ClosePaymentRequestV2Dto().apply {
+                    paymentTokens = activatedEvent.data.paymentNotices.map { it.paymentToken }
+                    outcome = transactionOutcome
+                    this.transactionId = transactionId.toString()
+                  }
+                ClosePaymentRequestV2Dto.OutcomeEnum.OK -> {
+                  val authCompletedEvent =
```

Comment:
Instead of find single transaction events I would refactor this method to reduce all transaction events and then check the reduced transaction object by doing something such as   
`val baseTransaction = reduceEvents(mono { transactionId.value() }, transactionsEventStoreRepository)`
and then performing check if the reduced transaction is an userCanceled and so on

Reply:
I refactored based on your comment.  👍